### PR TITLE
Bug/5358 Fix Dashboard Sharing Settings Modal position on mobile

### PR DESCRIPTION
**Pull request by jimmymadon**
## Summary

Addresses issue:

- #5358 

## Relevant technical choices

The solution within the IB did not fully fix the issue when we scrolled a tiny bit, so that only a part of the WP Admin Bar is scrolled. In this little edge case, the `top` and `height` values of the dialog box need to be set dynamically, based on the number of pixels actually scrolled. Also used flex to stretch instead of hardcoding heights. Could potentially clean up more of the height values on the `.mdc-dialog__container` that were recently added to develop - but not sure what they were fixing so have kept them for now.

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [x] Run the code.
- [x] Ensure the acceptance criteria are satisfied.
- [x] Reassess the implementation with the IB.
- [x] Ensure no unrelated changes are included.
- [x] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [x] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [x] Ensure the PR has the correct target branch.
- [x] Double-check that the PR is okay to be merged.
- [x] Ensure the corresponding issue has a ZenHub release assigned.
- [x] Add a changelog message to the issue.

### CODE REVIEW

File: assets/js/components/dashboard-sharing/DashboardSharingSettingsButton.js [hunk 1]
```diff
@@ -16,6 +16,10 @@
  * limitations under the License.
  */
 
+/**
+ * External dependencies
+ */
+import { useWindowScroll } from 'react-use';
```
Comment (asvinb) [inline assets/js/components/dashboard-sharing/DashboardSharingSettingsButton.js:22]: @jimmymadon Let's leave a line after all the external dependencies imports.

File: assets/js/components/dashboard-sharing/DashboardSharingSettingsButton.js [hunk 2]
```diff
@@ -87,6 +92,13 @@ export default function DashboardSharingSettingsButton() {
 					open={ dialogOpen }
 					onClose={ closeDialog }
 					className="googlesitekit-dialog googlesitekit-sharing-settings-dialog"
+					// On mobile, the dialog box's flexbox is set to stretch items within to cover
+					// the whole screen. But we have to move the box and adjust its height below the
+					// WP Admin bar of 46px which gradually scrolls off the screen.
+					style={ {
```
Comment (asvinb) [inline assets/js/components/dashboard-sharing/DashboardSharingSettingsButton.js:98]: @jimmymadon Nice! I'am thinking maybe we should not set the styles dynamically here on larger breakpoints, i.e we just set them on small breakpoints. we can use the existing `breakpoint` value we have to decide whether or not to add those styles.

**Review by asvinb — CHANGES_REQUESTED** [review]

**Comment by asvinb** [comment]
Nice catch @jimmymadon . You're right, the IB did not cater for when you scroll a little bit. I left a couple of comments, can you check them out and let me know what you think?

**Comment by makiost** [comment]
The changes from #5360 still work as they should 👍

**Comment by asvinb** [comment]
Thanks @jimmymadon !

**Review by techanvil — CHANGES_REQUESTED** [review]

**Pull request merged**